lib/jsdom/virtual-console.js

```diff
@@ -0,0 +1,80 @@
+"use strict";
+
+var EventEmitter = require("events").EventEmitter;
+
+var vConsole = {};
+
+var emitters = {};
+var windows = [];
+
+// Returns an EventEmitter that emits on any console events
+vConsole.getVirtualConsole = function(window) {
+  var virtualConsole = new EventEmitter();
+
+  saveEmitterForWindow(virtualConsole, window);
```

Comment:
The design here is way too clever, and without weak maps will leak memory. Just do `window._virtualConsole = virtualConsole`. Probably if it already exists, return it, instead of creating a new one.